kernel/share.c

```diff
@@ -282,20 +282,17 @@ static void free_file_table_entry(int fileno) {
 	file_table[fileno].filename[0] = '\0';
 }
 
-/* DOS 7 does not have read-only restrictions, MED 08/2004 */
-/*
-static int file_is_read_only(char FAR  *filename) {
-	union REGS regs;
-	struct SREGS sregs;
-
-	regs.x.ax = 0x4300;
-	sregs.ds = FP_SEG(filename);
-	regs.x.dx = FP_OFF(filename);
-	intdosx(&regs, &regs, &sregs);
-	if (regs.x.cflag) return 0;
-	return ((regs.h.cl & 0x19) == 0x01);
+static int file_is_read_only(char FAR *filename) {
```

Comment:
Please try this:
`static int file_is_read_only(__XFAR(char) filename)`
And move the `{` to new line. :)